packages/text-field/test/text-field.test.js

```diff
@@ -50,101 +49,42 @@ describe('text-field', () => {
           await assertAttrCanBeSet(prop, 'none');
         });
       });
 
       ['autocomplete', 'autocorrect', 'readonly', 'required'].forEach((prop) => {
-        it(`should set boolean attribute ${prop}`, async () => {
+        it(`should delegate ${prop} attribute to the input`, async () => {
           await assertAttrCanBeSet(prop, true);
           await assertAttrCanBeSet(prop, false);
         });
       });
     });
 
-    describe('clear button', () => {
-      it('should set clearButtonVisible to false by default', () => {
-        expect(textField.clearButtonVisible).to.be.false;
+    describe('internal', () => {
+      it('should store reference to the clear button element', () => {
+        expect(textField.clearElement).to.equal(textField.$.clearButton);
       });
 
-      it('should clear the value when clear button is clicked', async () => {
-        textField.clearButtonVisible = true;
-        textField.value = 'Foo';
-        await nextUpdate(textField);
-        textField.$.clearButton.click();
-        expect(textField.value).not.to.be.ok;
-      });
-
-      it('should clear the native input value when clear button is clicked', async () => {
-        textField.clearButtonVisible = true;
-        textField.value = 'Foo';
-        await nextUpdate(textField);
-        textField.$.clearButton.click();
-        expect(input.value).to.equal('');
-      });
-
-      it('should dispatch input event when clear button is clicked', async () => {
-        const inputSpy = sinon.spy();
-        textField.addEventListener('input', inputSpy);
-
-        textField.clearButtonVisible = true;
-        textField.value = 'Foo';
-        await nextUpdate(textField);
-
-        textField.$.clearButton.click();
-        expect(inputSpy.calledOnce).to.be.true;
-      });
-
-      it('should dispatch change event when clear button is clicked', async () => {
-        const changeSpy = sinon.spy();
-        textField.addEventListener('change', changeSpy);
-
-        textField.clearButtonVisible = true;
-        textField.value = 'Foo';
-        await nextUpdate(textField);
-
-        textField.$.clearButton.click();
-        expect(changeSpy.calledOnce).to.be.true;
-      });
-
-      it('should prevent default on clear button click', () => {
-        const event = new Event('click', { cancelable: true });
-        textField.$.clearButton.dispatchEvent(event);
-        expect(event.defaultPrevented).to.be.true;
-      });
-    });
-
-    describe('binding', () => {
-      it('default value should be empty string', () => {
-        expect(textField.value).to.be.equal('');
-      });
-
-      it('setting input value updates value', () => {
-        input.value = 'foo';
-        fire(input, 'input');
-        expect(textField.value).to.be.equal('foo');
+      it('should set ariaTarget property to the input element', () => {
+        expect(textField.ariaTarget).to.equal(textField.inputElement);
```

Comment:
The `ariaTarget` property was not covered at all (tests passed with it being removed).
I also added explicit tests for `focusElement` and `stateTarget` for consistency.